api/src/main/proto/stellarstation/api/v1/groundstation/plan-execution.proto

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package stellarstation.api.v1.groundstation;
+
+option go_package = "github.com/infostellarinc/go-stellarstation/api/v1/groundstation";
+
+option java_multiple_files = true;
+option java_outer_classname = "PlanExecution";
+option java_package = "com.stellarstation.api.v1.groundstation";
+
+// A life cycle event of a plan.
+message PlanExecutionEvent {
+  oneof Event {
+    // The plan is beginning. Sent at the plan's `start_time'.
+    PlanWillBegin begin = 2;
+
+    // The plan has ended. Sent when the plan's execution state is determined.
+    PlanDidEnd end = 3;
+  }
+}
+
+// An event notifying a plan is beginning.
+message PlanWillBegin {
+  // The desired state of the ground station's devices at each point of time between AOS and LOS.
+  repeated DeviceState state = 1;
+}
+
+// An event notifying a plan has ended.
+message PlanDidEnd {
+  // The execution status of the plan.
+  PlanExecutionStatus status = 1;
+}
+
+// A state of the ground station's devices.
+message DeviceState {
```

Comment:
What about everything else about the plan like antenna ID, satellite ID, channelset protocol, TLE, etc. etc. Don't external plugins need to know about this too?

Reply:
Good question.

I first thought it would make sense that at least one of those new messages have full plan data. I decided not to go that path. `DeviceState` is an descendant of `GroundStationStreamResponse`. The root message has plan id but not full plan data, which implies the full data should be retrieved by calling `ListPlans`.